metaflow/datatools/s3.py

```diff
@@ -562,17 +836,32 @@ def put_files(self, key_paths, overwrite=True):
         Put files to S3 in parallel.
 
         Args:
-            key_paths: (required) an iterator of (key, path) tuples.
+            key_paths: (required) an iterator of (key, path) tuples. Instead of
+                       (key, path) tuples, you can also pass any object that
+                       has the following properties 'key', 'value', 'content_type',
+                       'metadata' like the S3PutObject for example. 'key' and
+                       'value' are required but others are optional. In this case,
+                       'value' should point to the file path
             overwrite: (optional) overwrites the key with obj, if it exists
 
         Returns:
             a list of (key, S3 URL) tuples corresponding to the files sent.
         """
         def _check():
-            for key, path in key_paths:
+            for key_path in key_paths:
+                key = getattr(key_path, 'key', key_path[0])
+                path = getattr(key_path, 'value', key_path[1])
```

Comment:
I think using value instead of path here may be slightly confusing, how about: 

```suggestion
                path = getattr(key_path, 'path', key_path[1])
```

Reply:
I agree this is a bit confusing. I was trying to reuse the S3PutObject in all put calls. Let me see if I can come up with something a bit nicer.

Reply:
yeah, @romain-intel we could add `.path` in `S3PutObject`

Reply:
Done